src/components/UserSelector.tsx

```diff
@@ -1,24 +1,31 @@
-import React, { useContext, useEffect, useState } from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { getUsers } from '../features/users/users';
+import * as usersActions from '../features/users/usersSlice';
+import * as authorActions from '../features/author/authorSlice';
 
 type Props = {
   value: User | null;
-  onChange: (user: User) => void;
 };
 
 export const UserSelector: React.FC<Props> = ({
   // `value` and `onChange` are traditional names for the form field
   // `selectedUser` represents what actually stored here
   value: selectedUser,
-  onChange,
 }) => {
   // `users` are loaded from the API, so for the performance reasons
   // we load them once in the `UsersContext` when the `App` is opened
   // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
-  const [expanded, setExpanded] = useState(false);
+  const { users, expanded } = useAppSelector(state => state.users);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    getUsers()
+      .then(usersFromServer => dispatch(usersActions.set(usersFromServer)));
+  }, []);
```

Comment:
You do need to use getUser. You can you async thunk for requests and set data with extra reducers